homeassistant/components/flo/sensor.py

```diff
@@ -0,0 +1,213 @@
+"""Support for Flo Water Monitor sensors."""
+
+from typing import List, Optional
+
+from homeassistant.const import (
+    DEVICE_CLASS_PRESSURE,
+    DEVICE_CLASS_SIGNAL_STRENGTH,
+    DEVICE_CLASS_TEMPERATURE,
+    PRESSURE_PSI,
+    STATE_UNKNOWN,
+    TEMP_CELSIUS,
+    VOLUME_GALLONS,
+)
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.util.temperature import fahrenheit_to_celsius
+
+from .const import DOMAIN as FLO_DOMAIN, SIGNAL_WATER_CONSUMPTION_UPDATED
+from .device import FloDevice
+from .entity import FloEntity
+
+DEPENDENCIES = ["flo"]
+
+WATER_ICON = "mdi:water"
+GAUGE_ICON = "mdi:gauge"
+NAME_DAILY_USAGE = "Today's Water Usage"
+NAME_CURRENT_SYSTEM_MODE = "Current System Mode"
+NAME_FLOW_RATE = "Water Flow Rate"
+NAME_TEMPERATURE = "Water Temperature"
+NAME_WATER_PRESSURE = "Water Pressure"
+NAME_WIFI_SIGNAL_STRENGTH = "WiFi Signal Strength"
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Flo sensors from config entry."""
+    devices: List[FloDevice] = hass.data[FLO_DOMAIN]["devices"]
+    entities = []
+    entities.extend([FloDailyUsageSensor(device) for device in devices])
+    entities.extend([FloSystemModeSensor(device) for device in devices])
+    entities.extend([FloCurrentFlowRateSensor(device) for device in devices])
+    entities.extend([FloTemperatureSensor(device) for device in devices])
+    entities.extend([FloPressureSensor(device) for device in devices])
+    entities.extend([FloWifiStrengthSensor(device) for device in devices])
+    async_add_entities(entities, True)
+
+
+class FloDailyUsageSensor(FloEntity):
+    """Monitors the daily water usage."""
+
+    def __init__(self, device):
+        """Initialize the daily water usage sensor."""
+        super().__init__(
+            f"{device.mac_address}_daily_consumption", NAME_DAILY_USAGE, device
```

Comment:
Since we also pass in device, wouldn't it make sense for `FloEntitiy` to just prefix the passed in unique ID with mac_address ?

Reply:
so do the F string in the superclass? rather than here and just pass the `_daily_consumption` part to the superclass?

Reply:
`daily_consumption`, but yes.

Reply:
done